src/tsolvers/arraysolver/ArraySolver.cc

```diff
@@ -144,7 +144,8 @@ void ArraySolver::declareAtom(PTRef tr) {
                 if (logic.isArrayStore(term)) {
                     storeTerms.insert(eref);
                 }
-            } else if (logic.isArraySelect(term)) {
+            }
```

Comment:
What about adding a comment with e.g. `// Select does not necessarily yield array sort` ?